cpp/client/async_log_client.cc

```diff
@@ -163,39 +163,57 @@ void DoneGetEntries(UrlFetcher::Response* resp,
 
   for (int n = 0; n < jentries.Length(); ++n) {
     JsonObject entry(jentries, n);
-    if (!entry.Ok())
+    if (!entry.Ok()) {
       return done(AsyncLogClient::BAD_RESPONSE);
+    }
 
     JsonString leaf_input(entry, "leaf_input");
-    if (!leaf_input.Ok())
+    if (!leaf_input.Ok()) {
       return done(AsyncLogClient::BAD_RESPONSE);
+    }
 
     AsyncLogClient::Entry log_entry;
     if (Deserializer::DeserializeMerkleTreeLeaf(leaf_input.FromBase64(),
                                                 &log_entry.leaf) !=
-        Deserializer::OK)
+        Deserializer::OK) {
       return done(AsyncLogClient::BAD_RESPONSE);
+    }
 
     JsonString extra_data(entry, "extra_data");
-    if (!extra_data.Ok())
+    if (!extra_data.Ok()) {
       return done(AsyncLogClient::BAD_RESPONSE);
-
-    if (log_entry.leaf.timestamped_entry().entry_type() == ct::X509_ENTRY)
+    }
+
+    // This is an optional non-standard extension, used only by the log
+    // internally when running in clustered mode.
+    JsonString sct_data(entry, "sct");
+    if (sct_data.Ok()) {
+      unique_ptr<SignedCertificateTimestamp> sct(
+          new SignedCertificateTimestamp);
+      if (Deserializer::DeserializeSCT(sct_data.FromBase64(), sct.get()) !=
+          Deserializer::OK) {
+        return done(AsyncLogClient::BAD_RESPONSE);
+      }
+      log_entry.sct.reset(sct.release());
+    }
+
+    if (log_entry.leaf.timestamped_entry().entry_type() == ct::X509_ENTRY) {
       Deserializer::DeserializeX509Chain(extra_data.FromBase64(),
                                          log_entry.entry.mutable_x509_entry());
-    else if (log_entry.leaf.timestamped_entry().entry_type() ==
-             ct::PRECERT_ENTRY)
+    } else if (log_entry.leaf.timestamped_entry().entry_type() ==
+               ct::PRECERT_ENTRY) {
       Deserializer::DeserializePrecertChainEntry(
           extra_data.FromBase64(), log_entry.entry.mutable_precert_entry());
-    else
+    } else {
       LOG(FATAL) << "Don't understand entry type: "
                  << log_entry.leaf.timestamped_entry().entry_type();
+    }
 
-    new_entries.push_back(log_entry);
+    new_entries.emplace_back(std::move(log_entry));
   }
 
   entries->reserve(entries->size() + new_entries.size());
-  copy(new_entries.begin(), new_entries.end(), back_inserter(*entries));
+  move(new_entries.begin(), new_entries.end(), back_inserter(*entries));
```

Comment:
Somehow, this one doesn't need ```std::``` qualification?! One of those wacky resolution rules kicking in, I guess...